qiskit/mitigation/local_readout_mitigator.py

```diff
@@ -0,0 +1,269 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Readout mitigator class based on the 1-qubit local tensored mitigation method
+"""
+
+
+from typing import Optional, List, Tuple, Iterable, Callable, Union
+import numpy as np
+
+from qiskit.providers import Backend
+from qiskit.providers.models import BackendProperties
+from qiskit.result import Counts, QuasiDistribution
+from qiskit.exceptions import QiskitError
+from .base_readout_mitigator import BaseReadoutMitigator
+from .utils import counts_probability_vector, z_diagonal, str2diag
+
+
+class LocalReadoutMitigator(BaseReadoutMitigator):
+    """1-qubit tensor product readout error mitigator.
+    Mitigates expectation_value and quasi_probabilities.
+    The mitigator should either be calibrated using qiskit.experiments,
+    or calculated directly from the backend properties."""
+
+    def __init__(
+        self,
+        amats: Optional[List[np.ndarray]] = None,
+        backend: Optional[Union[Backend, BackendProperties]] = None,
+        qubits: Optional[Iterable[int]] = None,
+    ):
+        """Initialize a LocalReadoutMitigator
+        Args:
+            amats: Optional, list of single-qubit readout error assignment matrices.
+            backend: Optional, backend name.
+        """
+        if amats is None:
+            amats = self._from_backend(backend, qubits)
+        if qubits is None:
+            self._num_qubits = len(amats)
+            self._qubits = range(self._num_qubits)
+        else:
+            self._qubits = qubits
```

Comment:
For lookup later it might be good to convert qubits to a map for the index position in the assigment_mat, mitigation_mats like:
```
self._qubit_index = dict(zip(self._qubits, range(self._num_qubits))
```
Then later in the `assignment_matrix` and `mitigation_matrix` functions you can do something like
```
self._assignment_mats[self._qubit_index[qubit]]
```

Reply:
That's indeed very convenient. Done.